src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/executeStrategy/basicExecuteStrategy.ts

```diff
@@ -107,6 +118,9 @@ export class BasicExecuteStrategy implements ITerminalExecuteStrategy {
 			// execution requested
 			const finishedCommand = await onDone;
 
+			// Reset user input state after command execution completes
+			this._toolTerminal.receivedUserInput = false;
```

Comment:
This could live in `runInTerminalTool` after we `await execute()`, then the resetting is the same for all strategies